0025-pre-shared-key-2/0025-pre-shared-key-2.md

```diff
@@ -45,7 +50,7 @@ In PSKv2.0, request and response data is encrypted using [AES-256-GCM](https://e
 
 If subsequent versions support additional encryption algorithms, those details should be exchanged between the sender and receiver when they establish the pre-shared key. If a receiver attempts to decrypt an incoming packet but is unable to (perhaps because the sender is using an unsupported cipher), they SHOULD reject the incoming transfer with an `F06: Unexpected Payment` error.
```

Comment:
Maybe we could add another that's even more specific but the definition does say (emphasis added):

> The receiver was not expecting a payment like this (the memo and destination address don't make sense in that combination, for example if the **receiver does not understand the transport protocol used**)

FWIW, the implementation uses an `F06` to determine if it should try sending a legacy packet to support implementations of Draft 1 of this spec: https://github.com/interledgerjs/ilp-protocol-psk2/blob/master/src/sender.ts#L200

Reply:
I would vote for a new error code because I would consider this "invalid" as opposed to "unknown"

Reply:
Actually when you try to decrypt the packet and it doesn't work, you don't know if it was modified along the way or using an encryption algorithm you don't understand. All you know is that you weren't able to decrypt it.

Reply:
I would interpret "Unknown" to mean I got a payment that I wasn't expecting. I.e. I was able to parse it but it doesn't match up to anything I am expecting.

I would interpret "Invalid" to mean I got a payment that I can't even parse.

Reply:
Don't know what to say. That error code has meant what it says there since it was defined. The description pretty clearly indicates that it should be used for saying you don't understand the transport protocol.

If the user rejects the packet, for example because they weren't expecting the payment, that's treated as an `F99: Application Error` and the user can include some response in the PSK Response data to tell the sender what's up.

Small point but to me "invalid" sounds like "I parsed it and you sent something bad" like an HTTP 400 Bad Request error.

Reply:
Yeh, not worth making a big deal over.